### PR TITLE
Add tests and move files to have a better entry point

### PR DESCRIPTION
**Pull request by SaraVieira**
* Creates a helpers file with stuff that we need like `decodeURLParams`
* Adds tests to that file
* Creates constants.ts
* Creates folder with our styledModal
* exports everything in an index.tsx

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,9 +1,5 @@
 module.exports = {
-  extends: [
-    'react-app',
-    'prettier/@typescript-eslint',
-    'plugin:prettier/recommended',
-  ],
```
Comment (SaraVieira) [inline .eslintrc.js:6]: My eslint is fighting prettier

File: package.json [hunk 2]
```diff
@@ -14,10 +14,12 @@
     "start": "tsdx watch",
```
Comment (jfranciscosousa) [inline package.json:14]: I'm always triggering myself. Can we rename this to `dev`. I know it's 100% unrelated 😂

File: src/UrlModal.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect, Suspense, useCallback } from 'react';
+import { MODAL_KEY, PARAMS_KEY } from './constants';
+import { createURL, decodedUrlParams, encodeUrlParams } from './helpers';
+import useCustomEvent from './hooks/useCustomEvent';
+
+const routerPush = async (href: string) =>
+  window.history.pushState({ path: href }, '', href);
+
+export const openModal = ({
```
Comment (jfranciscosousa) [inline src/UrlModal.tsx:9]: Why not export these helpers too? Just curious, they are fine either way to me.

File: package.json (hunk 2 above)
Comment (SaraVieira) [reply package.json]: Honestly same, changed

File: src/UrlModal.tsx (hunk 3 above)
Comment (SaraVieira) [reply src/UrlModal.tsx:9]: Truu, should export the open modal

Done

Comment (SaraVieira) [reply src/UrlModal.tsx:9]: Wait i did bad things

**Review by jfranciscosousa — APPROVED** [review]

**Pull request merged**